bump luarocks packages

teto opened the pull request:
## Description of changes

WIP: bump luarocks package-set . Waiting to fix several luarocks upstream
before starting

- [x] lua-iconv: moved to lunarmodules with fixed rockspecs https://github.com/luarocks/luarocks-site/issues/199#issuecomment-1717484507
- [x] mimetypes: moved to lunarmodules with fixed rockspecs  
- [x] basexx: patch merged, we can use upstream again 
- [x] https://gitlab.com/peterbillam/pjb_lua/-/issues/2 (for now I've just removed it from the generated set and put directly in overrides)
- [x] https://github.com/JorjBauer/lua-cyrussasl/issues/6 wrong source, for now I've dropped it
- luarocks-nix: unstable-2023-02-26 -> unstable-2023-09-08
- [x] generateLuarocksConfig: add defaults to ease its usage
- wip

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

teto committed:
luarocks-nix: unstable-2023-02-26 -> unstable-2023-09-08

teto committed:
luaPackages: update csv files

- luaPackages.basexx: use upstream
- luaPackages.nui-nvim: fix name in csv file

alerque commented on pkgs/development/lua-modules/overrides.nix:
```diff
@@ -270,6 +270,13 @@ with prev;
 
   lua-iconv = prev.lua-iconv.overrideAttrs (oa: {
+    # rockspec is broken https://github.com/ittner/lua-iconv/issues/8
+    src = fetchFromGitHub {
```
This workaround should be obsolete, you can go back to just using the default LuaRock.

teto replied:
indeed, I removed it

teto committed:
luaPackages.readline: moved out from the generated set

teto committed:
luaPackages: update

teto committed:
update-luarocks-package: update script

else nixpkgs-review fails

teto committed:
mark lua-resty-session as broken

teto committed:
luaPackages.lua-ff-zlib: init

teto committed:
luaPackages.cyrussasl: remove because broken/old

and unused ?

teto commented:
current status: 
```
32 packages marked as broken and skipped:
lua51Packages.digestif lua51Packages.lmpfrlib lua52Packages.digestif lua52Packages.gitsigns-nvim lua52Packages.lmpfrlib lua52Packages.telescope-manix lua52Packages.telescope-nvim lua53Packages.gitsigns-nvim lua53Packages.telescope-manix lua53Packages.telescope-nvim lua54Packages.cyan lua54Packages.gitsigns-nvim lua54Packages.haskell-tools-nvim lua54Packages.http lua54Packages.ldoc lua54Packages.luaevent lua54Packages.lush-nvim lua54Packages.nvim-cmp lua54Packages.plenary-nvim lua54Packages.readline lua54Packages.sqlite lua54Packages.teal-language-server lua54Packages.telescope-manix lua54Packages.telescope-nvim lua54Packages.tl luaPackages.digestif luaPackages.gitsigns-nvim luaPackages.lmpfrlib luaPackages.telescope-manix luaPackages.telescope-nvim luajitPackages.digestif luajitPackages.lmpfrlib

12 packages failed to build:
lua51Packages.lua-resty-openidc lua51Packages.std-normalize luaPackages.lua-resty-openidc luaPackages.std-normalize lua53Packages.lua-resty-openidc lua53Packages.std-normalize lua54Packages.lua-resty-openidc lua54Packages.std-normalize luajitPackages.lua-resty-openidc luajitPackages.std-normalize nelua openrussian-cli

1889 packages built:
```

mrcjkb commented on pkgs/development/lua-modules/overrides.nix:
```diff
@@ -386,6 +387,11 @@ with prev;
     ];
   });
 
+  # lua-resty-session =  prev.lua-resty-session.overrideAttrs (oa: {
+  #   # lua_pack and lua-ffi-zlib are unpackaged, causing this package to not evaluate
+  #   meta.broken = true;
+  # });
```
It looks like it's already marked broken on `master`?

teto replied:
let's do this once this PR reaches master

mrcjkb commented:
> I am tempted to merge this as is, as among the build failures, I am pretty sure most of them already fail on master, and it's simpler to fix the leaf packages with small rebuilds on master.

Took a closer look at the broken packages (on `x86_64-linux`) on `master`:

- `lua*Packages.lua-resty-openidc` is marked as broken.
- `lua*Packages.std-normalize` builds.
- `nelua` builds (only has `luacheck` as a lua dependency).
- `openrussian-cli` isn't marked as broken, but fails to build.

teto committed:
openrussian-cli: mark as broken

openrussian-cli breaks on master previous to this PR with:

> patching sources
> updateAutotoolsGnuConfigScriptsPhase
> building
> build flags: SHELL=/nix/store/xdqlrixlspkks50m9b0mpvag65m3pf2w-bash-5.2-p15/bin/bash LUA=/nix/store/qx7i2wd5clvzislr1ix5j6173kryj1ln-lua-5.3.6-env/bin/lua LUAC=/nix/store/qx7i2wd5clvzislr1ix5j6173kryj1ln-lua-5.3.6-env/bin/luac
> echo "#!/nix/store/qx7i2wd5clvzislr1ix5j6173kryj1ln-lua-5.3.6-env/bin/lua" >openrussian
> /nix/store/qx7i2wd5clvzislr1ix5j6173kryj1ln-lua-5.3.6-env/bin/luac -o - openrussian.lua >>openrussian
> chmod a+x openrussian
> rm -f openrussian-sqlite3.db
> unzip -p openrussian-sql.zip openrussian.sql | awk -f ./mysql2sqlite - | sqlite3 openrussian-sqlite3.db
> awk: ./mysql2sqlite:214: warning: regexp escape sequence `\"' is not a known regexp operator
> memory
> sqlite3 openrussian-sqlite3.db -batch <postprocess.sql
> Parse error near line 21: no such column: '
>   TO bare_inflections SELECT word_id, REPLACE(temp, "'", "") AS bare FROM (  --
>                                       error here ---^

teto committed:
nelua: unstable-2023-01-21 -> unstable-2023-09-16

Commit:
luaPackages: updated the 10-22-2023

teto commented:
thanks for the thoughtful review, as a follow up I've:
- marked openrussian-cli as broken
- updated nelua
- tried to fix std._normalize but the build failed even then,  we were using an unstable version that is broken so instead I told it to use the stable version and opened https://github.com/nix-community/luarocks-nix/pull/32 to prepare for the next release. I've packaged ldoc anyway
- lua*Packages.lua-resty-openidc can be adjusted easily once this reaches master for ZHF, but I wanted to have this PR in staging before 23.11 release.

teto commented on pkgs/development/lua-modules/overrides.nix:
```diff
@@ -514,8 +520,19 @@ with prev;
 
-  readline = prev.readline.overrideAttrs (oa: {
-    propagatedBuildInputs = oa.propagatedBuildInputs ++ [ readline.out ];
+  readline = final.callPackage({ buildLuarocksPackage, fetchurl, luaAtLeast, luaOlder, lua, luaposix }:
+  buildLuarocksPackage ({
+    pname = "readline";
+    version = "3.2-0";
+    knownRockspec = (fetchurl {
+      url    = "mirror://luarocks/readline-3.2-0.rockspec";
+      sha256 = "1r0sgisxm4xd1r6i053iibxh30j7j3rcj4wwkd8rzkj8nln20z24";
+    }).outPath;
+    src = fetchurl {
+      url    = "http://www.pjb.com.au/comp/lua/readline-3.2.tar.gz";
+      sha256 = "1mk9algpsvyqwhnq7jlw4cgmfzj30l7n2r6ak4qxgdxgc39f48k4";
+    };
+
```
like for vimPlugins, the source is broken so I've stopped it from being generated. I've actually changed the rockspec url `url    = "http://www.pjb.com.au/comp/lua/readline-3.2.tar.gz";` to `url    = "http://pjb.com.au/comp/lua/readline-3.2.tar.gz";`
